test/e2e/windows/device_plugin.go

```diff
@@ -0,0 +1,124 @@
+limitations under the License.
+*/
+
+package windows
+
+import (
+//	"context"
+	"time"
+
+//	v1 "k8s.io/api/core/v1"
+//	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+//	appsv1 "k8s.io/api/apps/v1"
+//	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+
+	"github.com/onsi/ginkgo"
+)
+
+var _ = SIGDescribe("Device Plugin", func() {
+	f := framework.NewDefaultFramework("device-plugin")
+
+//	var cs clientset.Interface
+
+	ginkgo.BeforeEach(func() {
+		//Only for Windows containers
+		e2eskipper.SkipUnlessNodeOSDistroIs("windows")
+//		cs = f.ClientSet
+	})
+	ginkgo.It("should be able to create a functioning device plugin for Windows", func() {
+		ginkgo.By("creating Windows device plugin daemonset")
+//		dsName := "directx-device-plugin"
+//		daemonsetNameLabel := "daemonset-name"
+//		image := "directxplugin"
+//		mountName := "device-plugin"
+//		mountPath := "/var/lib/kubelet/device-plugins"
+//		privileged := true
+/*		labels := map[string]string{
+			daemonsetNameLabel: dsName,
+		}*/
+/*        	ds := &appsv1.DaemonSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: dsName,
+				Namespace: "kube-system",
+			},
+			Spec: appsv1.DaemonSetSpec{
+				Selector: &metav1.LabelSelector{
+					MatchLabels: labels,
+				},
+				Template: v1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Annotations: map[string]string{
+							"scheduler.alpha.kubernetes.io/critical-pod": "",
+						},
+						Labels: labels,
+					},
+					Spec: v1.PodSpec{
+						Tolerations: []v1.Toleration{
+							{
+								Key: "CriticalAddonsOnly",
+								Operator: "Exists",
+							},
+						},
+						Containers: []v1.Container{
+							{
+								Name:  "hostdev",
+								Image: image,
+								SecurityContext: &v1.SecurityContext{
+									Privileged: &privileged,
+								},
+								VolumeMounts: []v1.VolumeMount{
+									{
+										Name: mountName,
+										MountPath: mountPath,
+									},
+								},
+							},
+						},
+						Volumes: []v1.Volume{
+							{
+								Name: mountName,
+								VolumeSource: v1.VolumeSource{
+									HostPath: &v1.HostPathVolumeSource{
+										Path: mountPath,
+									},
+								},
+							},
+						},
+						NodeSelector: map[string]string{
+							"kubernetes.io/os": "windows",
+						},
+					},
+				},
+			},
+		}
+*/
+		ns := f.Namespace.Name
+//		ds, err := cs.AppsV1().DaemonSets(ns).Create(context.TODO(), ds, metav1.CreateOptions{})
+//		framework.ExpectNoError(err)
+
+		ginkgo.By("creating Windows testing Pod")
+		windowsPod := createTestPod(f, windowsBusyBoximage, windowsOS)
+		windowsPod = f.PodClient().CreateSync(windowsPod)
+
+		ginkgo.By("verifying device access in Windows testing Pod")
+		command := []string{"cmd.exe", "dxdiag", "/t", "dxdiag_output.txt", "&", "type", "dxdiag_output.txt"}
+		expectedString := "Todo: DirectX Version: DirectX 12"
```

Comment:
I suppose, we need to update this to pass test

Reply:
Yes.
Which should be easy to do once the test is running.

Reply:
so is fixed now test ran, but failed no ? https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/93948/pull-kubernetes-e2e-aks-engine-windows-gpu/1323857693282144256

Reply:
I think we need to remove the "todo:" from expextedString as well.

Reply:
> I think we need to remove the "todo:" from expextedString as well.

Of course.
But we first have to see what the expected string is on that node.

Reply:
> so is fixed now test ran, but failed no ? https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/93948/pull-kubernetes-e2e-aks-engine-windows-gpu/1323857693282144256

This run didn't include the /c fix yet.